src/Integration.UnitTests/Suppression/SuppressedIssuesFileStorageTests.cs

```diff
@@ -0,0 +1,372 @@
+            var issues = new List<SonarQubeIssue> { issue };
+
+            var logger = new Mock<ILogger>();
+
+            var directory = new Mock<IDirectory>();
+            var file = new Mock<IFile>();
+            file.Setup(f => f.ReadAllText(GetFilePath("projectKey"))).Returns(JsonConvert.SerializeObject(issues));
+            Mock<IFileSystem> fileSystem = CreateFileSystem(directory, file);
+
+            var fileStorage = new SuppressedIssuesFileStorage(logger.Object, fileSystem.Object);
+
+            var issuesGotten = fileStorage.Get("project:Key").ToList();
+
+            logger.Verify(l => l.WriteLine("SuppressedIssuesFileStorage: File name has illegal characters"), Times.Once);
+            Assert.AreEqual(0, issuesGotten.Count);
+        }
+
+        [TestMethod]
+        public void Update_Should_LogWhenAnErrorIsThrown()
+        {
+            SonarQubeIssue issue = CreateIssue("issueKey");
+            var issues = new List<SonarQubeIssue> { issue };
+
+            var logger = new Mock<ILogger>();
+
+            var directory = new Mock<IDirectory>();
+            var file = new Mock<IFile>();
+            file.Setup(f => f.WriteAllText(It.IsAny<string>(), It.IsAny<string>())).Throws(new Exception("Test Exception"));
+            Mock<IFileSystem> fileSystem = CreateFileSystem(directory, file);
+
+            var fileStorage = new SuppressedIssuesFileStorage(logger.Object, fileSystem.Object);
+
+
+            fileStorage.Update("projectKey", issues);
+
+            logger.Verify(l => l.WriteLine("SuppressedIssuesFileStorage Update Error: Test Exception"), Times.Once);
+        }
+
+        [TestMethod]
+        public void Get_Should_LogWhenAnErrorIsThrown()
+        {
+            SonarQubeIssue issue = CreateIssue("issueKey");
+            var issues = new List<SonarQubeIssue> { issue };
+
+            var logger = new Mock<ILogger>();
+
+            var directory = new Mock<IDirectory>();
+            var file = new Mock<IFile>();
+            file.Setup(f => f.ReadAllText(GetFilePath("projectKey"))).Throws(new Exception("Test Exception"));
+            Mock<IFileSystem> fileSystem = CreateFileSystem(directory, file);
+
+            var fileStorage = new SuppressedIssuesFileStorage(logger.Object, fileSystem.Object);
+
+            var issuesGotten = fileStorage.Get("projectKey").ToList();
+
+            logger.Verify(l => l.WriteLine("SuppressedIssuesFileStorage Get Error: Test Exception"), Times.Once);
+            Assert.AreEqual(0, issuesGotten.Count);
+        }
+
+        private static Mock<IFileSystem> CreateFileSystem(Mock<IDirectory> directory, Mock<IFile> file)
+        {
+            var fileSystem = new Mock<IFileSystem>();
+            fileSystem.SetupGet(fs => fs.File).Returns(file.Object);
+            fileSystem.SetupGet(fs => fs.Directory).Returns(directory.Object);
+            return fileSystem;
+        }
+
+        private static SonarQubeIssue CreateIssue(string issueKey)
```

Comment:
I used only one field to differentiate between records. Adding more fields would not serve any purpose other than testing the Serializer.